Dockerfile

```diff
@@ -1,13 +1,20 @@
 FROM golang:1.22-alpine as builder
 RUN apk add --no-cache git
-WORKDIR /go/src/github.com/google/cloud-android-orchestrator
+WORKDIR /go/src/github.com/google/cloud-android-orchestration
 COPY . .
 RUN GO111MODULE=on CGO_ENABLED=0 GOOS=linux go build \
       -trimpath \
-      -o /app \
+      -o /cloud_orchestrator \
       cmd/cloud_orchestrator/main.go
 
-FROM gcr.io/distroless/base
-COPY --from=builder /app /app
+FROM gcr.io/distroless/base as runner-base
```

Comment:
Let's containerize the binary only and not the configuration file to be able to use the same image using different configuration files. The configuration file will be manage when running the container i.e `docker run --volume /config:/config --env CONFIG_FILE='/config/conf.toml cuttlefish-cloud-orchestration:latest`

Reply:
Since we decided not to add configuration files into the docker image today, is there any existing public(for any users) deployment for this docker image? Unless, I would rather to push this docker image to AR repository where docker image containing HO located.

Reply:
I'm not aware any public deployment of this docker image. We should keep and maintain a docker image artifact only if requested with proper use case, not as a nice thing to have.

Reply:
We want to provide a guide about using cvdr + CO (with docker instance manager) for a partner to use CHD on top of their ARM server.

Reply:
I see, let's have the guide in `scripts/on-premises/single-server/README.md` first to better understand what are we trying to achieve.

Reply:
Let's rebrand the cvdr + CO (with docker instance manager) deployment as "On Premises Single Server" or "Single Server" for short, rather than just docker to avoid confusion, as docker is used for containerizing the binary on every deployment plus is also an instance manager implementation.

Reply:
Some questions, since having some unclear things before proceeding this PR. Want to reduce gaps between us.
- Do we need to keep `scripts/docker/*`? I think renaming the directory from `scripts/docker/` into `scripts/on-premises/single-server` looks fine. It won't have different roles I think.
- I want to move `docs/docker.md` into `script/on-premises/single-server/` with some edit, sounds good?
- What's the term `single-server` for? I'm fine for the either way whether containing it or not in the dir path. But I want to know the detailed reason since I'm not certain yet. Does it mean 'single CO server'? Also, do you have a mind for another model which is not fit on `single-server`?
- Just for confirmation, having only one docker build target & explicitly specifying config from the outside(e.g. `--volume` option) is still valid?

Reply:
> Do we need to keep scripts/docker/*? I think renaming the directory from scripts/docker/ into scripts/on-premises/single-server looks fine. It won't have different roles I think.

We can remove `scipts/docker/*`. Yes, we will be moving the same content to `scripts/on-premises/single-server/*`.

>I want to move docs/docker.md into script/on-premises/single-server/ with some edit, sounds good?

Yes.

>What's the term single-server for? I'm fine for the either way whether containing it or not in the dir path. But I want to know the detailed reason since I'm not certain yet. Does it mean 'single CO server'? Also, do you have a mind for another model which is not fit on single-server?

I'm using the term `single-server` as the on-premises deployment using the current Docker Instance manager in the documentation works only on a single machine. The CO service and the Docker Engine both needs to be running on the same machine. Yes, in the future, we should have an on-premises scalable solution, where we are going to have a different Instance Manager implementation that can manage docker containers at scale (in multiple machine clusters). I'm thinking of an Instance Manager which is the client of the Kubernetes Resource Management API for example, still need to design this instance manager.


> Just for confirmation, having only one docker build target & explicitly specifying config from the outside(e.g. --volume option) is still valid?

Yes, I think we should have a uniform way to containerize the CO binary in our code base, but different ways to run it.

Reply:
Adopted all requirements in this comment.